mdmon.c

```diff
@@ -394,9 +394,7 @@ int main(int argc, char *argv[])
 		/* launch an mdmon instance for each container found */
 		mdstat = mdstat_read(0, 0);
 		for (e = mdstat; e; e = e->next) {
-			if (e->metadata_version &&
-			    strncmp(e->metadata_version, "external:", 9) == 0 &&
-			    !is_subarray(&e->metadata_version[9])) {
+			if (is_mdstat_ent_subarray(e)) {
```

Comment:
incorrect check,
is_mdstat_ent_subarray expect volume to be subarray, but this if statement expect oposite state

Reply:
good catch! As I said, it requires to be extra careful and I failed :)